checkbox-ng/plainbox/impl/ctrl.py

```diff
@@ -133,12 +135,31 @@ def get_dependency_set(self, job, job_list=None):
         )
         return result
 
+    def add_before_deps(self, job, job_map):
+        """
+        Add a all "before" references declared in a job to the corresponding
+        jobs as an "after" dependency in the before_references set.
+
+        If a job (B) has a "before" field, we add this job as an "after"
+        dependency to the job (A).
```

Comment:
This is not very clear, since job A and job B are not defined.

You could show a very simple job definition, and how this mechanism turns

```
id:job-a

id: job-b
before: job-a
```
into

```
id:job-a
after: job-b

id: job-b
```